qemu/tests/balloon_thp.py

```diff
@@ -0,0 +1,62 @@
+import time
+
+from avocado.utils import process
+from virttest.staging import utils_memory
+
+
+def run(test, params, env):
+    """
+    Memory balloon with thp
+
+    1. Boot up a guest with balloon support, record memory fragement
+    2. Make fragement in guest with tmpfs
+    3. check the memory fragement with proc system, should increase
+    4. Do memory balloon the memory size ballooned should be a legal value
+    5. Check the memory fragement with proc system
+    6. Try to make the memory compact
+    7. Check the memory status in guest
+    8. Check the fragement in system should decrease
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def check_buddy_info(bf, af, increase=True):
+        """
+        Compare if buddy info increase or decrease
+        :param bf: buddy info dict before
+        :param af: buddy info dict after
+        :param increase: Expect the value increase or decrease
+        :return True if meet expectation else False
+        """
+        for key, value in bf.items():
```

Comment:
Let's only check the first item in af and bf, which means the number of 4k fragment.